addons/pos_loyalty/static/src/app/services/pos_store.js

```diff
@@ -410,10 +422,10 @@ patch(PosStore.prototype, {
             delete opt.price_unit;
         }
 
+        vals.isRewardProductLine = opt?.isRewardProductLine;
```

Comment:
```js
        const result = await super.addLineToCurrentOrder(vals, opt, configure);
        result.uiState.isRewardProductLine = opt.isRewardProductLine;
```

Reply:
BTW in `addLineToCurrentOrder` of point_of_sale we should return 
```
return to_merge_orderline || line;
```

otherwise the returned object can be wrong.